app/src/main/java/com/example/firstapplication/UserAdapter.kt

```diff
@@ -39,10 +41,22 @@ class UserAdapter(
         deleteUserButton.setOnClickListener {
             remove(user.id)
         }
+
+        holder.itemView.setOnClickListener {
+            openUserDetailsActivity(it.context, user)
+        }
+    }
+
+    private fun openUserDetailsActivity(context: Context, user: User) {
```

Comment:
Taka logika nie powinna być w adapterze. Otwieranie nowej aktywności to nie jest jego odpowiedzialność :)
Lepiej byłoby poinformować MainActivity poprzez callback (lambdę) że kliknięto w kafelek usera.
I wtedy w aktywności MainActivity możesz otworzyć UserDetailsActivity